syntax = "proto3";

package greeting;

service UserService
{
  rpc CreateUser(UserRequest) returns (UserResponse) {}
  rpc UpdateUser(UserRequest) returns (UserResponse) {}
  rpc DeleteUser(DeleteUserRequest) returns (UserResponse) {}
  rpc GetTicker(GetTickerRequest) returns (Ticker) {}
}

message UserRequest
{
  string email = 1;
  string ticker = 2;
}

message DeleteUserRequest
{
  string email = 1;
}

message UserResponse
{
  string message = 1;
}

message GetTickerRequest
{
    string email = 1;
}

message Ticker
{
    string ticker = 1;
}